traits/traits_listener.py

```diff
@@ -1122,10 +1194,11 @@ def parse_item(self, *, terminator, deferred, handler_type):
                 last = result
                 while last.next is not None:
                     last = last.next
-                last.next = lg = ListenerGroup(items=[next, result])
+                lg = ListenerGroup(items=[next, result])
```

Comment:
General comment: there are a number of simplifications that would be possible if we didn't have to deal with  recursion and cycles; the cycles add substantial complication for a feature that's rarely used. Leaving the cycles out of the new `observe` machinery seems as though it was a Good Thing.